src/gh/components/DF_joint_segmentator/code.py

```diff
@@ -0,0 +1,69 @@
+#! python3
+
+import Rhino
+import diffCheck
+from diffCheck import diffcheck_bindings
+from diffCheck import df_cvt_bindings as df_cvt
+import diffCheck.df_util
+import scriptcontext as sc
+
+ABSTOL = Rhino.RhinoDoc.ActiveDoc.ModelAbsoluteTolerance
+
+def main(i_clusters, i_joints, i_joint_ids):
+
+    if len(i_joints) != len(i_joint_ids):
+        raise ValueError("The number of joints and joint ids must be the same.")
+    
+    if len(i_clusters) == 0:
+        raise ValueError("No clusters given.")
+
+    if not isinstance(i_clusters[0], Rhino.Geometry.PointCloud):
+        raise ValueError("The input clusters must be PointClouds.")
+    
+    if not isinstance(i_joints[0], Rhino.Geometry.Mesh):
+        raise ValueError("The input joints must be convertible to Meshes.")
+        
+
+    # prepping the reference meshes
+    n_joints = max(i_joint_ids) + 1
+    joints = [ [] for i in range(n_joints) ]
+    for face, id in zip(i_joints, i_joint_ids):
+        face.Subdivide()
+        face.Faces.ConvertQuadsToTriangles()
+        joints[id].append(df_cvt.cvt_rhmesh_2_dfmesh(face))
+
+    joint_clouds = []
+    registrations = []
+    joint_segments = []
+    df_clouds = [df_cvt.cvt_rhcloud_2_dfcloud(cluster) for cluster in i_clusters]
+
+    # for each joint, find the corresponding clusters and merge them, generate a reference point cloud, and register the merged clusters to the reference point cloud
+    for joint in joints:
+
+        # create the reference point cloud
+        joint_cloud = diffcheck_bindings.dfb_geometry.DFPointCloud()
+
+        for face in joint:
+            face_cloud = face.sample_points_uniformly(1000)
```

Comment:
proposal: this could become a function in the `DFFace` object like `to_dfcloud()` just as `to_mesh()`

Reply:
but now, sample_points_uniformly(n_points) is a member of DFMesh. I don't get understand what would be different in what you propose ?